MediaBrowser.Api/Playback/MediaInfoService.cs

```diff
@@ -95,14 +100,36 @@ public class MediaInfoService : BaseApiService
             _networkManager = networkManager;
             _mediaEncoder = mediaEncoder;
             _userManager = userManager;
-            _json = json;
             _authContext = authContext;
         }
 
         public object Get(GetBitrateTestBytes request)
         {
-            var bytes = new byte[request.Size];
-            return ResultFactory.GetResult(null, bytes, "application/octet-stream");
+            const int MaxSize = 10_000_000;
+
+            var size = request.Size;
+
+            if (size <= 0)
+            {
+                throw new ArgumentException($"The requested size can't be equal or smaller than 0.", nameof(request));
+            }
+
+            if (size > MaxSize)
+            {
+                throw new ArgumentException($"The requested size can't be larger than the max allowed value ({MaxSize}).", nameof(request));
+            }
+
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(size);
+            try
+            {
+                // ArrayPool<byte>.Shared.Rent doesn't guarantee that the returned buffer is zeroed
+                Array.Fill<byte>(buffer, 0);
```

Comment:
No need to, it would actually better be filled by random stuff to imitate a movie which isn't easily well `gzip`-able by a reverse proxy. I would actually say let's allocate some random 10MB junk on first request and keep it forever...

Reply:
But you should make sure that you don't just send what's in the memory, because that not good at all for security

Reply:
You can use the pseudo random `Random` class to fill some buffer, it's fast enough.